packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -483,57 +253,68 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		}
 	);
 
-	if ( isLaunchPage ) {
-		return (
-			<LaunchPagePlanFeatureActionButton
-				planSlug={ planSlug }
-				planTitle={ planTitle }
-				priceString={ priceString }
-				isStuck={ isStuck }
-				isLargeCurrency={ !! isLargeCurrency }
-				onCtaClick={ onCtaClick }
-			/>
-		);
-	}
-	if ( isInSignup ) {
-		return (
-			<SignupFlowPlanFeatureActionButton
-				planSlug={ planSlug }
-				planTitle={ planTitle }
-				priceString={ priceString }
-				isStuck={ isStuck }
-				isLargeCurrency={ !! isLargeCurrency }
-				hasFreeTrialPlan={ !! freeTrialPlanSlug }
-				onCtaClick={ onCtaClick }
-				onFreeTrialCtaClick={ onFreeTrialCtaClick }
-				planActionOverrides={ planActionOverrides }
-			/>
-		);
-	}
+	const {
+		primary: { callback, text, status },
+		postButtonText,
+	} = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		availableForPurchase,
+		billingPeriod,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
+		planSlug,
+		planTitle,
+		priceString,
+		cartItemForPlan,
+		currentPlanBillingPeriod,
+		selectedStorageAddOn,
+	} );
 
-	return (
-		<LoggedInPlansFeatureActionButton
-			planSlug={ planSlug }
-			availableForPurchase={ availableForPurchase }
-			onCtaClick={ onCtaClick }
-			currentSitePlanSlug={ currentSitePlanSlug }
-			buttonText={ buttonText }
-			planActionOverrides={ planActionOverrides }
-			priceString={ priceString }
-			isStuck={ isStuck }
-			isLargeCurrency={ !! isLargeCurrency }
-			isMonthlyPlan={ isMonthlyPlan }
-			planTitle={ planTitle }
-			storageOptions={ storageOptions }
-		/>
-	);
-};
+	const {
+		primary: { callback: freeTrialCallback, text: freeTrialText },
+	} = useAction( {
```

Comment:
We won't be needing this second call eventually (so also answers your question below regarding the interface). We'd just pass `freeTrialPlanSlug` separately and get both actions back (the `{ primary, secondary }` Action). 

something along these lines

Reply:
Makes sense 🙂